source/Crew.cpp

```diff
@@ -0,0 +1,242 @@
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Crew.h"
+#include "GameData.h"
+
+using namespace std;
+
+// Load definition of a crew member
+void Crew::Load(const DataNode &node)
+{
+	// Set the id of this crew member so we know that we have loaded it.
+	if(node.Size() >= 2)
+		id = node.Token(1);
+	
+	for(const DataNode &child : node)
+	{
+		if(child.Size() >= 2)
+		{
+			// The crew members' display name (plural, Title Case)
+			if(child.Token(0) == "name")
+				name = child.Token(1);
+			// Each valid ship has at least this many of the crew member
+			else if(child.Token(0) == "minimum per ship")
+				minimumPerShip = max((int)child.Value(1), 0);
+			// The number of credits paid daily while parked (minimum 0)
+			else if(child.Token(0) == "parked salary")
+				parkedSalary = max((int)child.Value(1), 0);
+			// Every nth crew member on the ship will be this crew member
+			else if(child.Token(0) == "population per member")
+				populationPerMember = max((int)child.Value(1), 0);
+			// The number of credits paid daily (minimum 0)
+			else if(child.Token(0) == "salary")
+				salary = max((int)child.Value(1), 0);
+			else
+				child.PrintTrace("Skipping unrecognized attribute:");
+		}
+		// If true, the crew member will not appear on escorts
+		else if(child.Token(0) == "avoids escorts")
+			avoidsEscorts = true;
+		// If true, the crew member will not appear on the flagship
+		else if(child.Token(0) == "avoids flagship")
+			avoidsFlagship = true;
+		else
+			child.PrintTrace("Skipping incomplete attribute:");
+	}
+}
+
+
+
+int64_t Crew::CalculateSalaries(
+	const vector<shared_ptr<Ship>> &ships,
+	const Ship * &flagship,
+	const bool includeExtras
+)
+{
+	int64_t totalSalaries = 0;
+	
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		totalSalaries += Crew::SalariesForShip(
+			ship,
+			ship.get() == flagship,
+			includeExtras
+		);
+	}
+	
+	return totalSalaries;
+}
+
+
+
+int64_t Crew::CostOfExtraCrew(
+	const vector<shared_ptr<Ship>> &ships,
+	const Ship * &flagship
+)
+{
+	// Calculate with and without extras and return the difference.
+	return Crew::CalculateSalaries(ships, flagship, true)
+		- Crew::CalculateSalaries(ships, flagship, false);
+}
+
+
+
+int64_t Crew::NumberOnShip(
+	const Crew &crew,
+	const shared_ptr<Ship> &ship,
+	const bool isFlagship,
+	const bool includeExtras
+)
+{
+	int64_t count = 0;
+	
+	// If this is the flagship, check if this crew avoids the flagship.
+	if(isFlagship && crew.AvoidsFlagship())
+		return count;
+	// If this is an escort, check if this crew avoids escorts.
+	if(!isFlagship && crew.AvoidsEscorts())
+		return count;
+	
+	const int64_t countableCrewMembers = includeExtras
+		? ship->Crew()
+		: ship->RequiredCrew();
+	
+	// Apply the per-ship minimum.
+	 count = min(crew.MinimumPerShip(), countableCrewMembers);
```

Comment:
Is this aligned right?

Reply:
Yes. If it looks weird, it's because of the ternary above it.

Reply:
No, it looks weird because there is a space-character too much before the c of count.

Reply:
Oh well spotted!